### ISSUE TITLE
Upgrade to the new version of hardhat. Error: "Table has empty fields"

### ISSUE THREAD

**Issue by maxmartynov** [aurora-is-near/hardhat-storage-layout#29]
**Problem**
I receive the error `Table has empty fields`

**How to reproduce**
```
smart-contracts git:(main)$ yarn hardhat check
yarn run v1.22.19
$ /Users/max/work/<project>/smart-contracts/node_modules/.bin/hardhat check
Table has empty fields
✨  Done in 1.06s.
```

**Description**
The problem is the plugin is looking `stateVariables` in the files `/artifacts/contracts/**/*.json`, but all the required info is now located in the `/artifacts/build-info/*.json` files. I'm not sure, perhaps one of the new versions of `hardhat` causes this problem. All the contracts are valid, pass tests, compile correctly and work as expected.

**Location of the problem**
Here is the variable/piece of code that should be updated - https://github.com/aurora-is-near/hardhat-storage-layout/blob/main/src/storageLayout.ts#L55

```
...
        const storage =
          artifactJsonABI.data.output?.contracts?.[sourceName]?.[contractName]
            ?.storageLayout?.storage;
...            
```

**My config**
Chip: Apple M1 Pro
Node: v18.12.0
NPM: v8.19.2

```
# package.json
{
  ...
  "devDependencies": {
    "@ethersproject/abi": "5.7.0",
    "@ethersproject/providers": "5.7.2",
    "@nomicfoundation/hardhat-chai-matchers": "^1.0.0",
    "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
    "@nomicfoundation/hardhat-toolbox": "^1.0.1",
    "@nomiclabs/hardhat-ethers": "2.2.1",
    "@nomiclabs/hardhat-etherscan": "3.1.4",
    "@nomiclabs/hardhat-web3": "2.0.0",
    "@openzeppelin/contracts": "4.7.3",
    "@typechain/ethers-v5": "^10.1.0",
    "@typechain/hardhat": "^6.1.2",
    "@types/chai": "^4.2.0",
    "@types/dotenv": "^8.2.0",
    "@types/mocha": "^9.1.0",
    "@types/node": ">=12.0.0",
    "chai": "4.3.7",
    "dotenv": "16.0.1",
    "ethers": "5.7.2",
    "hardhat": "2.12.5",
    "hardhat-contract-sizer": "^2.6.1",
    "hardhat-gas-reporter": "1.0.9",
    "hardhat-storage-layout": "^0.1.7",
    "prettier": "^2.8.2",
    "solhint": "^3.3.7",
    "solidity-coverage": "^0.8.2",
    "ts-node": ">=8.0.0",
    "typechain": "^8.1.1",
    "typescript": ">=4.5.0",
    "web3": "^1.8.1"
  }
}
```

**Comment by nlukic97** [comment]
After installing this plugin, make sure to run:
`npx hardhat clean`

`npx hardhat compile`

`npx hardhat check`

Replace **npx hardhat** with **yarn** if you are using it.

See the following images:
![image](https://user-images.githubusercontent.com/56925545/218765238-abb39fa9-b815-4a0a-823f-4e0141597a59.png)

When the contracts are not compiled, the tables will be empty. But when I run the three commands from above, it works.

Which version of hardhat are you using?

**Comment by maxmartynov** [comment]
Hm. It's weird because I checked the `/artifacts` directory before creating the issue and the built files were in there. Moreover, I launched tests and they could not be passed without compiling the contracts.

But yes, your instructions solved the problem:

```
yarn hardhat clean
yarn hardhat compile
yarn hardhat check
```

I've already changed the environment and maybe the problem was in something else, but I cannot check and reproduce it again. Let's consider the problem solved.

Thank you!
